### PR TITLE
[MRG+1] FEED_EXPORT_FIELDS option

### LINKED ISSUE #1050: Issue with CsvItemExporter when used as Feed Exporter with a crawler that yields more than one type of Item
My project includes a spider that yields several different types of Items. All my items inherit from a 'DefaultItem' that has fields such as 'url', 'time_crawled' and 'group_id'.

When defining `FEED_FORMAT = 'csv'`, the feed exporter uses the fields of the first type of Item yielded as the csv's header/column names and then for subsequent yielded items (despite the Item's type) tries to export those items with the fields of the first yielded item.

Since all of the items have 'url', 'time_crawled', 'group_id' and fields that may share the same names, those columns in the CSV get exported but otherwise the column for particular field is empty.

```tim@lindev:~/kkc$ scrapy -v
Scrapy 0.24.4 - project: kkc```

### PR DESCRIPTION
**Pull request by kmike**
As discussed in https://github.com/scrapy/scrapy/pull/1081, this option is nice to have if we're yielding items as dicts. This also fixes #1050.

There were no tests for FeedExporter class, so I've added some.

### CODE REVIEW

File: scrapy/utils/testproc.py [hunk 1]
```diff
@@ -1,13 +1,15 @@
+from __future__ import absolute_import
```
Comment (pablohoffman) [inline scrapy/utils/testproc.py]: just curious about why this is needed and whether this is related to this pull request

**Comment by pablohoffman** [comment]
looking good!

File: scrapy/utils/testproc.py (hunk 1 above)
Comment (kmike) [inline scrapy/utils/testproc.py]: I went back and forth when writing FeedExporter tests; first tried to subclass tests from test_commands, then tried to use utilities from scrapy.utils.testproc, but in the end settled on another solution. That's why https://github.com/scrapy/scrapy/commit/973c31f73595c73fac9b825ce7587d93d7055546 happened.

`from __future__ import absolute_import` is always good to have, I'd add it to all Scrapy modules :) It makes imports compatible in Python 2.x and Python 3.x and prevents stupid errors when a file in the current folder shadows stdlib module or other module in sys.path.

Comment (kmike) [inline scrapy/utils/testproc.py]: Should I remove the first commit?

Comment (pablohoffman) [inline scrapy/utils/testproc.py]: No, it's fine. I was just curious.

**Pull request merged**